crypto/rsa/rsa_pk1.c

```diff
@@ -270,6 +276,232 @@ int RSA_padding_check_PKCS1_type_2(unsigned char *to, int tlen,
     return constant_time_select_int(good, mlen, -1);
 }
 
+
+static int ossl_rsa_prf(OSSL_LIB_CTX *ctx,
+                        unsigned char *to, int tlen,
+                        const char *label, int llen,
+                        const unsigned char *kdk,
+                        uint16_t bitlen)
+{
+    int pos;
+    int ret = -1;
+    uint16_t iter = 0;
+    uint16_t be_iter;
+    HMAC_CTX *hmac = NULL;
+    EVP_MD *md = NULL;
+    unsigned char hmac_out[32];
+    unsigned int md_len;
+
+    if (tlen * 8 != bitlen) {
+        ERR_raise(ERR_LIB_RSA, ERR_R_INTERNAL_ERROR);
+        return ret;
+    }
+
+    bitlen = htons(bitlen);
```

Comment:
I am not sure htons is available everywhere - there is the OPENSSL_NO_SOCK define for example. I'd recommend using the IS_LITTLE_ENDIAN define from internal/endian.h to do the byte swap if necessary.

Reply:
I went with explicit serialisation, this way I don't need to run tests on a big endian system and compiler should handle it in a clever way still (as it's just 2 bytes)